static/app/utils/theme/theme.tsx

```diff
@@ -1108,6 +1127,12 @@ export type IconSize = Size;
 export type Aliases = typeof lightAliases;
 export type ColorOrAlias = keyof Aliases | Color;
 
+export type StrictCSSObject = {
+  [K in keyof CSSProperties]?: CSSProperties[K]; // Enforce standard CSS properties
+} & Partial<{
+  [key: `&${string}`]: StrictCSSObject; // Allow pseudo-selectors
```

Comment:
Technically this allows any nested selector

```suggestion
  [key: `&${string}`]: StrictCSSObject; // Allow nested selectors
```